exec/kubernetes_exec_manager.go

```diff
@@ -222,9 +223,22 @@ func (*KubernetesExecManager) Resize(id int, cols uint, rows uint) error {
 	return nil
 }
 
-func getById(id int) *model.MachineExec {
-	defer machineExecs.mutex.Unlock()
+// getK8sAPI return k8s api object.
+func (manager *KubernetesExecManager) getK8sAPI(userToken string) (k8s8API *client.K8sAPI, err error) {
```

Comment:
Could you elaborate on why did you put here?

Does it makes sense to give k8sClientProvider whole machine-exec and encapsulate token evaluation there...
```
k8sAPI, err := manager.getK8sAPI(machineExec)
```